osgp/protocol-adapter-dlms/osgp-protocol-adapter-dlms/src/test/java/org/opensmartgridplatform/adapter/protocol/dlms/domain/commands/AttributeAddressHelperServiceTest.java

```diff
@@ -0,0 +1,135 @@
+    private static final AttributeAddress ACTIVE_ENERGY_EXPORT_RATE_2 = new AttributeAddress(3, new ObisCode(1, 0, 2, 8, 2, 255), 2);
+
+    private GetPeriodicMeterReadsCommandExecutor executor;
+
+    @Mock
+    private DlmsMessageListener listener;
+
+    @Mock
+    private DlmsHelperService helperService;
+
+    @Mock
+    private AttributeAddressHelperService service;
+
+    @Before
+    public void setUp() {
+        this.service = new AttributeAddressHelperService(this.helperService);
+    }
+
+    @Test
+    public void testSelectedValuesForPeriodicMeterReads() throws Exception {
+        final List<AttributeAddress> expectedAttributeAddressesDaily = Arrays.asList(
+                CLOCK, AMR_PROFILE_STATUS,
+                ACTIVE_ENERGY_IMPORT_RATE_1, ACTIVE_ENERGY_IMPORT_RATE_2,
+                ACTIVE_ENERGY_EXPORT_RATE_1, ACTIVE_ENERGY_EXPORT_RATE_2);
+
+        final List<AttributeAddress> expectedAttributeAddressesMonthly = Arrays.asList(
+                CLOCK,
+                ACTIVE_ENERGY_IMPORT_RATE_1, ACTIVE_ENERGY_IMPORT_RATE_2,
+                ACTIVE_ENERGY_EXPORT_RATE_1, ACTIVE_ENERGY_EXPORT_RATE_2);
+
+        List<DataObject> selectedValues;
+
+        selectedValues = this.getAttributeAddresses(PeriodTypeDto.INTERVAL, true);
+        Assertions.assertThat(selectedValues.size()).isEqualTo(0);
+
+        selectedValues = this.getAttributeAddresses(PeriodTypeDto.DAILY, true);
+        this.assertListContainsObisCodes(selectedValues, expectedAttributeAddressesDaily);
+
+        selectedValues = this.getAttributeAddresses(PeriodTypeDto.MONTHLY, true);
+        this.assertListContainsObisCodes(selectedValues, expectedAttributeAddressesMonthly);
+
+        selectedValues = this.getAttributeAddresses(PeriodTypeDto.INTERVAL, false);
+        Assertions.assertThat(selectedValues.size()).isEqualTo(0);
+
+        selectedValues = this.getAttributeAddresses(PeriodTypeDto.DAILY, false);
+        Assertions.assertThat(selectedValues.size()).isEqualTo(0);
+
+        selectedValues = this.getAttributeAddresses(PeriodTypeDto.MONTHLY, false);
+        Assertions.assertThat(selectedValues.size()).isEqualTo(0);
+    }
+
+    private List<DataObject> getAttributeAddresses(final PeriodTypeDto periodType,
+            final boolean isSelectingValuesSupported) throws Exception {
+
+        // Setup mocks
+        when(this.helperService.getAMRProfileDefinition()).thenReturn(DataObjectDefinitions.getAMRProfileDefinition());
+        when(this.helperService.getClockDefinition()).thenReturn(DataObjectDefinitions.getClockDefinition());
+
+        // Setup captor to capture the selected values
+        final ArgumentCaptor<DataObject> dataObjectSelectedValuesCaptor = ArgumentCaptor.forClass(DataObject.class);
+        when(this.helperService.getAccessSelectionTimeRangeParameter(any(), any(), dataObjectSelectedValuesCaptor.capture())).thenReturn(DataObject.newNullData());
```

Comment:
Don't use argument captors for stubbing. See other comment for details.

Reply:
Ok, changed